server/cli/launch.py

```diff
@@ -183,4 +188,9 @@ def launch(
         f = open(devnull, "w")
         sys.stdout = f
 
-    server.app.run(host=host, debug=debug, port=port, threaded=True)
+    try:
+        server.app.run(host=host, debug=debug, port=port, threaded=True)
+    except OSError as e:
+        if e.errno == errno.EADDRINUSE:
+            raise click.ClickException(f"{e} Port is in use, please specify an open port using the --port flag.")
```

Comment:
Personally, I prefer chained exceptions.

```suggestion
            raise click.ClickException(f"Port is in use, please specify an open port using the --port flag.") from e
```

Reply:
fixed